### PR TITLE
5 - repository pattern

### PR DESCRIPTION
**Pull request by SuciuB**

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -6,12 +6,30 @@
...
+            "justMyCode": false,  
+            "requireExactSource": false 
+        },
+
+        {
+            "name": ".NET Core Launch (ParkingApplication.Api)",
+            "type": "coreclr",
+            "request": "launch",
+            "preLaunchTask": "build",
+            "program": "${workspaceFolder}/ParkingApplication.Api/bin/Debug/net8.0/ParkingApplication.Api.dll",  // Adjust the path if needed
```
Comment (Dyu13) [inline .vscode/launch.json]: remove the comments

File: ParkingApplication.api/Controllers/ParkingController.cs [hunk 2]
```diff
@@ -7,7 +7,7 @@
 namespace ParkingApplication.Controllers;
 
     [ApiController]
-    [Route("api/[controller]")]
+    [Route("api/parked-car")]
```
Comment (Dyu13) [inline ParkingApplication.api/Controllers/ParkingController.cs:10]: Is this right indented?

File: ParkingApplication.api/Controllers/ParkingController.cs [hunk 3]
```diff
@@ -21,24 +21,24 @@ public ParkedCarsController (IParkingService parkingService)
...
         {
-            _parkingService.AddToParking(car.CarId, car.CarNumber);
+            _parkingService.AddToParking(car.Id, car.CarNumber);
             return Ok();
         }
         catch (Exception ex)
         {
             return BadRequest(ex.Message);
-        }
+        }
```
Comment (Dyu13) [inline ParkingApplication.api/Controllers/ParkingController.cs]: remove space after braces

File: ParkingApplication.api/Interfaces/IParkingService.cs [hunk 4]
```diff
@@ -9,9 +9,12 @@ namespace ParkingApplication.Api.Interfaces
     public interface IParkingService
     {
 
-        List<ParkingModel> GetParkedCar { get; }
+        //List<ParkingModel> GetParkedCar { get; }
         public int MaxSlots { get; set; }   
         public void AddToParking(int id, string carNumber);
         public bool ExitParking(string carNumber);
+        void AddParking(ParkingModel parkingModel);
```
Comment (Dyu13) [inline ParkingApplication.api/Interfaces/IParkingService.cs:14]: use private if they are private methods

File: ParkingApplication.api/Models/AccountModel.cs [hunk 5]
```diff
@@ -7,14 +7,12 @@ namespace ParkingApplication.Api.Models;
 
 public class AccountModel
 {
-    public int AccountId { get; set; }
```
Comment (Dyu13) [inline ParkingApplication.api/Models/AccountModel.cs]: why we don't have an accountId anymore?

File: ParkingApplication.api/Models/ParkingModel.cs [hunk 6]
```diff
@@ -7,14 +7,14 @@ namespace ParkingApplication.Api.Models;
 
 public class ParkingModel
 {
-    public int CarId { get; set; }
+    public int Id { get; set; }
```
Comment (Dyu13) [inline ParkingApplication.api/Models/ParkingModel.cs:10]: don't you need a relationship with the user here?

File: ParkingApplication.api/Services/ParkingService.cs [hunk 7]
```diff
@@ -12,31 +13,52 @@ public class ParkingService : IParkingService
...
+            _commandRepository.Create(Car);
 
         }
     }
 
     public bool ExitParking(string carNumber)
     {
-        var parkedCar = GetParkedCar.Find(car => car.CarNumber == carNumber);
+        var getCars = _queryRepository.GetAll();
+        var parkedCar = getCars.Find(car => car.CarNumber == carNumber);
```
Comment (Dyu13) [inline ParkingApplication.api/Services/ParkingService.cs]: you could move this into repository and have a GetByCarNumber method

**Review by Dyu13 — CHANGES_REQUESTED** [review]

File: .vscode/launch.json (hunk 1 above)
Comment (SuciuB) [reply .vscode/launch.json]: Done!

File: ParkingApplication.api/Controllers/ParkingController.cs (hunk 3 above)
Comment (SuciuB) [reply ParkingApplication.api/Controllers/ParkingController.cs]: done

File: .vscode/launch.json (hunk 1 above)
Comment (SuciuB) [reply .vscode/launch.json]: done

File: ParkingApplication.api/Interfaces/IParkingService.cs (hunk 4 above)
Comment (SuciuB) [reply ParkingApplication.api/Interfaces/IParkingService.cs:14]: they are not private.

File: ParkingApplication.api/Models/ParkingModel.cs (hunk 6 above)
Comment (SuciuB) [reply ParkingApplication.api/Models/ParkingModel.cs:10]: Done!

File: ParkingApplication.api/Models/AccountModel.cs (hunk 5 above)
Comment (SuciuB) [reply ParkingApplication.api/Models/AccountModel.cs]: Done!

File: ParkingApplication.api/Interfaces/IParkingService.cs (hunk 4 above)
Comment (SuciuB) [reply ParkingApplication.api/Interfaces/IParkingService.cs:14]: Not Private! Its done!

File: ParkingApplication.api/Services/ParkingService.cs (hunk 7 above)
Comment (SuciuB) [reply ParkingApplication.api/Services/ParkingService.cs]: Done!

**Pull request merged**